### PR TITLE
STDTC-43: add jest/react-testing-library tests for FileUploader

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose

https://issues.folio.org/browse/STDTC-43
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by vashjs — APPROVED** [review]

File: lib/FileUploader/FileUploader.test.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+} from '@testing-library/react';
+
+import '../../test/jest/__mock__';
+import '../../test/jest/__new_mock__';
+
+import { FileUploader } from './FileUploader';
+
+describe('FileUploader', () => {
+  describe('rendering file uploader with children passed as is', () => {
+    const titleText = 'Title text';
```
Comment (uladislausamets) [inline lib/FileUploader/FileUploader.test.js:13]: move mocks out of test

File: lib/FileUploader/FileUploader.test.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        isDropZoneActive={false}
+        onDragEnter={onDragEnterMock}
+        onDragLeave={onDragLeaveMock}
+        onDrop={onDropMock}
+      >
+        <div data-testid="children">Children</div>
+      </FileUploader>
+    );
+
+    const fileUploader = (
```
Comment (uladislausamets) [inline lib/FileUploader/FileUploader.test.js:61]: why do you need this? `renderFileUploader` and `render(fileUploader);` works the same way

File: lib/FileUploader/FileUploader.test.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        dispatchEvt(fileInput, 'dragenter', data);
+        await flushPromises(fileUploader, container);
+
+        expect(onDragEnterMock).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('rendering file uploader with active drop area', () => {
+    const renderFileUploaderActive = () => render(
```
Comment (uladislausamets) [inline lib/FileUploader/FileUploader.test.js:109]: You can pass props in `renderFileUploader`

File: lib/FileUploader/FileUploader.test.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply lib/FileUploader/FileUploader.test.js]: done.

File: lib/FileUploader/FileUploader.test.js (hunk 2 above)
Comment (UladzislauKutarkin) [reply lib/FileUploader/FileUploader.test.js]: done.

File: lib/FileUploader/FileUploader.test.js (hunk 3 above)
Comment (UladzislauKutarkin) [reply lib/FileUploader/FileUploader.test.js]: done.

**Review by uladislausamets — APPROVED** [review]

**Pull request merged**